### DISCUSSION TITLE
How to run several simulations in a single GAMA headless instance ?

### DISCUSSION THREAD

**Discussion by corentinjuvigny** [gama-platform/gama#89] [Q&A] [answered]
As part of a research project, I am currently using Gama for evaluating the quality of some solutions of an algorithm (written in c++). Thus, I use the headless mode as I don't care about visual interface. But although it actually works, I have an issue in term of computation time because each time I evaluate a solution (which is done many many times) Gama needs to reload Java, which takes around 8-10 seconds.

Is there a mean to load Java once and for all in order to prevent the execution from taking hours (currently around 95% of the execution is due to the loading of Java) ? It would help me a lot.

![image](https://user-images.githubusercontent.com/49539269/150993652-649ee5f1-c3b9-4cdd-8110-b99dc63ac9ef.png)

Thanks.

**Comment by RoiArthurB** [comment, marked as answer]
Hello, 

I'm not sure how you are using GAMA headless, but you probably feed an XML file to the script as explained [in the documentation](https://gama-platform.org/wiki/Headless-mode-for-dummies).

By default, because GAMA can't predict what combination you want to run, the command `./gama-headless.sh -xml expName file.gaml file.xml` just generate a single `<Simulation>` (with parameters' default value). 

The trick to do what you want is that the `<Simulation>` tag isn't the root tag of your XML file. Actually, you can write (manually for now) several simulations in the root `<Experiment_plan>` tag and GAMA will run them one after the other. 

Even better, in GAMA, 1 simulation runs on 1 core, so if you give several simulations to your headless, it will run them in parallel (if you have 8 cores, you can run 8 simulations in parallel!). You can choose the number of cores with the `-hpc` flag (e.g `./gama-headless.sh -hpc 3 ./file.xml ./outputFolder`), by default it takes all your computer's cores. However, remember that you're giving the RAM to GAMA, not to a simulation (so if you run 3 simulations on 3 GiB of RAM, each simulation will have around 1 GiB of RAM to run).

I hope it answered your question and I let you close this issue if so :smile:
